lnutil/dlclib.go

```diff
@@ -0,0 +1,481 @@
+package lnutil
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"math/big"
+
+	"github.com/adiabat/btcd/btcec"
+	"github.com/adiabat/btcd/chaincfg/chainhash"
+	"github.com/adiabat/btcutil"
+
+	"github.com/adiabat/btcd/txscript"
+	"github.com/adiabat/btcd/wire"
+)
+
+// DlcContractStatus is an enumeration containing the various statuses a contract can have
+type DlcContractStatus int
+
+const (
+	ContractStatusDraft        DlcContractStatus = 0
+	ContractStatusOfferedByMe  DlcContractStatus = 1
+	ContractStatusOfferedToMe  DlcContractStatus = 2
+	ContractStatusDeclined     DlcContractStatus = 3
+	ContractStatusAccepted     DlcContractStatus = 4
+	ContractStatusAcknowledged DlcContractStatus = 5
+	ContractStatusActive       DlcContractStatus = 6
+	ContractStatusClosed       DlcContractStatus = 7
+)
+
+// scalarSize is the size of an encoded big endian scalar.
+const scalarSize = 32
+
+// DlcContract is a struct containing all elements to work with a Discreet Log Contract. This struct is stored in the database
+type DlcContract struct {
+	Idx                                      uint64                           // Index of the contract for referencing in commands
+	PeerIdx                                  uint32                           // Index of the peer we've offered the contract to or received the contract from
+	PubKey                                   [33]byte                         // Key of the contract
+	CoinType                                 uint32                           // Coin type
+	OracleA, OracleR                         [33]byte                         // Pub keys of the oracle
+	OracleTimestamp                          uint64                           // The time we expect the oracle to publish
+	Division                                 []DlcContractDivision            // The payout specification
+	OurFundingAmount, TheirFundingAmount     int64                            // The amounts either side are funding
+	OurChangePKH, TheirChangePKH             [20]byte                         // PKH to which the contracts funding change should go
+	OurFundMultisigPub, TheirFundMultisigPub [33]byte                         // Pubkey used in the funding multisig output
+	OurPayoutPub, TheirPayoutPub             [33]byte                         // Pubkey to which the contracts are supposed to pay out
```

Comment:
probably makes sense to also add a payout pubkey hash, and perhaps rename PayoutPub to PayoutBase (as it will not be used directly as a public key)